packages/babel-plugin-transform-destructuring/src/index.ts

```diff
@@ -560,6 +113,7 @@ export default declare((api, options) => {
         path.ensureBlock();
 
         const block = node.body;
+        // @ts-expect-error: ensureBlock ensures that node.body is a BlockStatement
```

Comment:
Could you check if changing the `ensureBlock` signature to this helps?
```js
function ensureBlock<T extends  t.Loop | t.WithStatement | t.Function | t.LabeledStatement | t.CatchClause>(
  this: NodePath<T>,
): assert this is NodePath<T & { node: t.BlockStatement }>
```

Reply:
Unfortunately we can't add type assertion without breaking change because:
1) `ensureBlock` returns a `t.Node` while assertion predicates requires that the function returns `void`.
2) `this`-type guards can only be added to a class method, though we can move it back to `index` or put it in a separate file and extends an interface (type with assertions) instead.